ts/screens/onboarding/PinScreen.tsx

```diff
@@ -84,400 +53,180 @@ const styles = StyleSheet.create({
-            onFulfill={this.onPinConfirmFulfill}
-            ref={pinpad => (this.pinConfirmComponent = pinpad)} // eslint-disable-line
-            buttonType={"light"}
-            onDeleteLastDigit={this.onPinConfirmRemoveLastDigit}
-          />
-        </React.Fragment>
-      );
-    }
-  }
+  const handlePinBlur = React.useCallback(() => {
+    setIsPinDirty(true);
+  }, [setIsPinDirty]);
 
-  // The Content of the Screen
-  public renderContent(pinState: PinState) {
-    return (
-      <Content>
-        {this.renderContentHeader(pinState)}
-        {this.renderErrorDescription()}
-        {this.renderCodeInput(pinState)}
-        {this.renderDescription()}
-      </Content>
-    );
-  }
+  const handlePinConfirmationBlur = React.useCallback(() => {
+    setIsPinConfirmationDirty(true);
+  }, [setIsPinConfirmationDirty]);
 
-  // Render the description for the different states
-  public renderDescription() {
-    return (
-      <Text alignCenter={true} style={styles.description}>
-        {I18n.t("onboarding.unlockCode.pinInfo")}
-      </Text>
+  const showRestartModal = React.useCallback(() => {
+    showModal(
+      <AlertModal
+        message={I18n.t("profile.main.pagoPaEnvironment.alertMessage")}
+      />
     );
-  }
+  }, [showModal]);
 
-  public renderContinueButton(pinState: PinState) {
-    if (pinState.state !== "PinConfirmed") {
-      return undefined;
+  const handleSubmit = React.useCallback(() => {
+    if (!isFormValid) {
+      return;
     }
 
-    return (
-      <View accessible={true} ref={this.continueButtonRef}>
-        <ButtonDefaultOpacity
-          block={true}
-          primary={true}
-          disabled={false}
-          onPress={() => this.setPin(pinState.pin)}
-        >
-          <Text>{I18n.t("global.buttons.continue")}</Text>
-        </ButtonDefaultOpacity>
-        <View spacer={true} />
-      </View>
-    );
-  }
+    const typedPin = pin as PinString;
 
-  // The Footer of the Screen
-  public renderFooter(pinState: PinState) {
-    return (
-      <View style={styles.footerContainer}>
-        {/* the actual footer must be wrapped in this container in order to keep a white background below the safe area */}
-        <View footer={true}>
-          {this.renderContinueButton(pinState)}
-
-          {pinState.state !== "PinUnselected" && (
-            <React.Fragment>
-              <ButtonDefaultOpacity
-                block={true}
-                bordered={true}
-                onPress={() => this.onPinReset()}
-                // small={true} TODO: it should be height 40 and text 16 - conflict with message cta style
-              >
-                <Text>{I18n.t("onboarding.unlockCode.reset")}</Text>
-              </ButtonDefaultOpacity>
-            </React.Fragment>
-          )}
-        </View>
-      </View>
-    );
-  }
+    void safeSetPin(typedPin)
+      .fold(
+        error => {
+          handleSendAssistanceLog(assistanceTool, `setPin error ${error}`);
```

Comment:
Depending on the behavior of [Keychain.setGenericPassword](https://github.com/pagopa/io-app/pull/3939/files#diff-339040eb4b756ab97806623e8fd7080a8da4c9ab5ff8e7538996605f2d8e3c0eR28) the error may contain sensitive data. I think it is worth checking it before logging.

Reply:
You're right! 🙏 [Currently we are not sending the error at all](https://github.com/pagopa/io-app/blob/e028b5ef159d9e15bb71bb639b96e7c94800d50a/ts/screens/onboarding/PinScreen.tsx#L450-L461), but just `setPin error`. It should be fixed here: https://github.com/pagopa/io-app/pull/3939/commits/7c6bf6f0f901ecd7a15b7ce38b8d1b8357c35427